memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateConversationRequest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.memory.action.conversation;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+import static org.opensearch.ml.common.conversation.ConversationalIndexConstants.APPLICATION_TYPE_FIELD;
+import static org.opensearch.ml.common.conversation.ConversationalIndexConstants.META_NAME_FIELD;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.core.common.io.stream.InputStreamStreamInput;
+import org.opensearch.core.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentParser;
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+public class UpdateConversationRequest extends ActionRequest {
+    private String conversationId;
+    private Map<String, Object> updateContent;
+
+    private static final Set<String> allowedList = new HashSet<>(Arrays.asList(META_NAME_FIELD, APPLICATION_TYPE_FIELD));
+
+    @Builder
+    public UpdateConversationRequest(String conversationId, Map<String, Object> updateContent) {
+        this.conversationId = conversationId;
+        this.updateContent = filterMapContent(updateContent);
+    }
+
+    public UpdateConversationRequest(StreamInput in) throws IOException {
+        super(in);
+        this.conversationId = in.readString();
+        this.updateContent = filterMapContent(in.readMap());
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(this.conversationId);
+        out.writeMap(this.getUpdateContent());
```

Comment:
What if updateContent is null?

Reply:
I added a check in the filterUpdateContent(Map<String, Object> updateContent) method. If updateContent is null, it returns a new Map with empty content to avoid null pointer. Also the ActionRequestValidationException validate() method will throw an exception if updateContent is null.